python/opae.admin/opae/admin/tools/fpgasupdate.py

```diff
@@ -597,16 +787,27 @@ def main():
         LOG.debug('Found secure device for PAC '
                   '%s', pac.pci_node.pci_address)
 
-        try:
-            stat, mesg = update_fw(args, pac)
-        except SecureUpdateError as exc:
-            stat, mesg = exc.errno, exc.strerror
-        except KeyboardInterrupt:
-            if TMPFILE and os.path.isfile(TMPFILE):
-                os.remove(TMPFILE)
-            cancel = sec_dev.find_one(os.path.join('update', 'cancel'))
-            cancel.value = 1
-            stat, mesg = 1, 'Interrupted'
+        if use_ioctl:
+            with pac.fme:
+                with sec_dev as fd:
+                    try:
+                        stat, mesg = update_fw_ioctl(fd, args.file, pac)
+                    except SecureUpdateError as exc:
+                        stat, mesg = exc.errno, exc.strerror
+                    except KeyboardInterrupt:
+                        fcntl.ioctl(fd, IOCTL_FPGA_IMAGE_LOAD_CANCEL)
```

Comment:
Need a try: here to guard against ioctl exceptions. If the ioctl() fails, an OSError exception is raised.